tutorials/appengine-pusher/index.md

```diff
@@ -0,0 +1,449 @@
+   clients. 
+    
+## Before you begin  
+
+1.  Create a project in the [Google Cloud Platform Console](https://console.cloud.google.com/).
+2.  Install the [Google Cloud SDK](https://cloud.google.com/sdk/) and run:
+    ```
+       gcloud init
+    ```
+    If this is your first time creating an App engine application:
+    ```
+       gcloud app create
+    ```
+
+## Getting started with Pusher service
+
+Pusher offers the ability to create realtime bi-directional functionality via WebSockets to web and
+ mobile apps, or any other Internet connected device. 
+Using WebSockets, Pusher is able to deliver server-side events to clients.
+Pusher also offers  HTTP REST API based server-side SDKs to enable sending events from server to a 
+public or secured channel.
+
+### Setting up your Pusher account.
+
+Follow these steps:
+
+1. [Create](https://pusher.com/signup) a Pusher account.
+1. Once signed in, you will be directed to a [dashboard](https://dashboard.pusher.com/).
+   The dashboard provides a convenient way to retrieve application settings, view errors
+   and a console to debug calls to your application.
+1. Using the left panel on the dashboard, create a new application and note down the cluster,
+   application id, key and secret.
+1. Update [appengine-web.xml](java/src/main/webapp/WEB-INF/appengine-web.xml) with your Pusher
+   account credentials.
+   
+### Pusher SDKs
+
+Pusher provides a [range of libraries in different languages](https://pusher.com/docs/libraries):
+
+- Their REST SDKs provide the ability to authenticate clients and 
+  publish HTTP events from your server.
+- The WebSocket SDKs handle subscription.
+
+### Channels
+
+A [channel](https://pusher.com/docs/client_api_guide/client_channels) is automatically created 
+when an application publishes or subscribes to the channel by name. 
+They do not need to be explicitly created or deleted.
+There are three types of channels:
+
+- Public : Any one can join the channel using the channel name and requires no authentication
+- Private: Server-side authentication is enforced, channel names must be prefixed with "private-"
+- Presence: Server-side authentication is enforced, channel names must be prefixed with "presence-",
+   and all members can view who have connected / disconnected from the channel.
+
+### Events
+
+An [Event](https://pusher.com/docs/client_api_guide/client_events) is a  message with a named type.
+Custom event handlers can be attached to a given event type. 
+This allows for efficient event routing in the clients.
+Note: A subscriber will receive all messages published over a channel.
+
+Events may be trigged by the user or Pusher.
+In case of Pusher triggered events on a channel, the event name is
+prefixed with `pusher:`, for example: `pusher:subscription-succeeded`.
+
+## Chat application
+The sample application demonstrates presence channels in Pusher for a chat application:
+View complete source code [here](java).
```

Comment:
Did you mean `(/java)` ? will that correctly point to the sub-folder java ? I could n't get it to work on testing.